pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -1,2 +1,161 @@
+import { expect } from 'chai';
+import { BigNumber } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { MINUTE, advanceTime, currentTimestamp } from '@balancer-labs/v2-helpers/src/time';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import WeightedPool from '@balancer-labs/v2-helpers/src/models/pools/weighted/WeightedPool';
+
+import { itBehavesAsWeightedPool } from './BaseWeightedPool.behavior';
+
 // eslint-disable-next-line @typescript-eslint/no-empty-function
-describe('LiquidityBootstrappingPool', function () {});
+describe('LiquidityBootstrappingPool', function () {
+  describe('as a 2-token weighted pool', () => {
+    itBehavesAsWeightedPool(2);
+  });
+
+  describe('as a 3-token weighted pool', () => {
+    itBehavesAsWeightedPool(3);
+  });
+
+  let trader: SignerWithAddress, lp: SignerWithAddress, owner: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, lp, trader, owner] = await ethers.getSigners();
+  });
+
+  let tokens: TokenList;
+
+  sharedBeforeEach('deploy tokens', async () => {
+    tokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+    await tokens.mint({ to: [lp, trader], amount: fp(200) });
+  });
+
+  let pool: WeightedPool;
+  const weights = [fp(0.3), fp(0.55), fp(0.1), fp(0.05)];
+  const initialBalances = [fp(0.9), fp(1.8), fp(2.7), fp(3.6)];
+
+  sharedBeforeEach('deploy pool', async () => {
+    const params = { tokens, weights, owner, lbp: true, swapEnabledOnStart: true };
+    pool = await WeightedPool.create(params);
+  });
+
+  /*const initializePool = () => {
+    sharedBeforeEach('initialize pool', async () => {
+      await pool.init({ initialBalances, recipient: lp });
+    });
+  };*/
+
+  describe('weights', () => {
+    it('sets token weights', async () => {
+      const normalizedWeights = await pool.getNormalizedWeights();
+
+      // Need to decrease precision
+      expect(normalizedWeights).to.equalWithError(pool.normalizedWeights, 0.0001);
+    });
+  });
+
+  describe('public swap', () => {
+    sharedBeforeEach('deploy tokens', async () => {
+      const action = await actionId(pool.instance, 'setPublicSwap');
+      await pool.vault.grantRole(action, owner);
+    });
+
+    it('Owner can pause swaps', async () => {
+      await pool.setPublicSwap(owner, false);
+
+      expect(await pool.instance.swapEnabled()).to.be.false;
+
+      await expect(pool.swapGivenIn({ in: 1, out: 0, amount: fp(0.1) })).to.be.revertedWith('SWAPS_PAUSED');
+
+      await pool.setPublicSwap(owner, true);
+
+      expect(await pool.instance.swapEnabled()).to.be.true;
+    });
+
+    it('LP cannot pause swaps', async () => {
+      await expect(pool.setPublicSwap(lp, false)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
```

Comment:
What do you think about splitting these up in a few more concrete tests? e.g.

```
context when the sender is the owner
   it owner can pause
   it pause emits an event
   context when paused
       before each pause
       it swap reverts
       it owner can unpause
       it non owner cannot unpause
       it unpause emits an event
context when the sender is not the owner
   it non owner cannot unpaused
```
Also I'd use a signer called `other` instead of `lp` for these things, as being an LP doesn't really have anything to do with this.